gamemodes/terrortown/gamemode/client/cl_targetid.lua

```diff
@@ -53,54 +52,37 @@ local materialBaseOverlay = Material("vgui/ttt/dynamic/sprite_base_overlay")
 ---
 -- Function that handles the drawing of the overhead roleicons, it does not check whether
 -- the icon should be drawn or not, that has to be handled prior to calling this function
+-- @param Player client The client entity
 -- @param Player ply The player to receive an overhead icon
--- @param Material ricon
--- @param Color rcolor
+-- @param Material iconRole The role icon that will be drawn
+-- @param Color colorRole The role color for the background
 -- @realm client
-function DrawOverheadRoleIcon(ply, ricon, rcolor)
-	local client = LocalPlayer()
-	if ply == client then
-		return
-	end
+function DrawOverheadRoleIcon(client, ply, iconRole, colorRole)
+	local pos = ply:GetBoneMatrix(ply:LookupBone("ValveBiped.Bip01_Head1")):GetTranslation()
```

Comment:
What if a model does not have this? Can we just add a fallback position then?

Reply:
good idea, I will add the old code as a fallback here